Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -926,13 +924,21 @@ void MatrixWorkspace::setYUnitLabel(const std::string &newLabel) {
 * TODO: For example: ????
 * @return whether workspace is a distribution or not
 */
-const bool &MatrixWorkspace::isDistribution() const { return m_isDistribution; }
+bool MatrixWorkspace::isDistribution() const {
+  return getSpectrum(0).yMode() == HistogramData::Histogram::YMode::Frequencies;
```

Comment:
Is it worth having a size check here? I don't know if we can have zero spectrum in a MatrixWorkspace, but this might break? I haven't checked, but maybe you are relying on a size check in `getSpectrum`.